### PR TITLE
feat: case stayed scheduler with camuda (CMC-638)

### PR DESCRIPTION
**Pull request by swalker125**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CMC-638

### Change description ###

New Camunda bmpn that runs on a schedule.
External task client configuration class.
Case Stayed Listener class.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by dharmendrak — DISMISSED** [review]

File: build.gradle [hunk 1]
```diff
@@ -306,9 +310,13 @@ dependencies {
     exclude group: 'org.junit.vintage', module: 'junit-vintage-engine'
   }
 
-  testImplementation 'org.mockito:mockito-inline:3.5.10'
+  testImplementation group: 'org.camunda.bpm.extension.springboot', name: 'camunda-bpm-spring-boot-starter-test', version: '2.2.0'
+
+  testImplementation group: 'com.h2database', name: 'h2', version: '1.4.200'
```
Comment (pd-dudkiewicz) [inline build.gradle:315]: is this used anywhere?

File: src/main/java/uk/gov/hmcts/reform/unspec/config/ExternalTaskListenerConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+    private final String baseUrl;
+
+    public ExternalTaskListenerConfiguration(@Value("${feign.client.config.remoteRuntimeService.url}") String baseUrl) {
+        this.baseUrl = baseUrl;
+    }
+
+    //TODO: define a workerId, max tasks and a sensible backOffStrategy - i.e how long should I wait between long
+    // polling, currently it is the default exponential back off strategy (double the value until it reaches 60000ms).
+    // Default max tasks is 10, and lock time per task is 20000L. Do we need to think about authentication?
```
Comment (pd-dudkiewicz) [inline src/main/java/uk/gov/hmcts/reform/unspec/config/ExternalTaskListenerConfiguration.java]: Can we get rid of it before merging? If the questions are not answered they should be replaced by ticket reference imho.

File: build.gradle (hunk 1 above)
Comment (swalker125) [reply build.gradle:315]: h2 is requried for createStandaloneInMemProcessEngineConfiguration on line 33 of BmpnBaseTest

https://github.com/hmcts/unspec-service/blob/d6b870374a9a2a195817ca23bcb25717ddaff512/src/test/java/uk/gov/hmcts/reform/unspec/bpmn/BpmnBaseTest.java#L33

Diving into the code `this.jdbcUrl = "jdbc:h2:mem:camunda";`

File: src/main/java/uk/gov/hmcts/reform/unspec/config/ExternalTaskListenerConfiguration.java (hunk 2 above)
Comment (swalker125) [reply src/main/java/uk/gov/hmcts/reform/unspec/config/ExternalTaskListenerConfiguration.java]: yeah I agree this is too much, actually chatting through some of this with Chris at the moment 👍

Comment (swalker125) [reply src/main/java/uk/gov/hmcts/reform/unspec/config/ExternalTaskListenerConfiguration.java]: pulled this out to this ticket https://tools.hmcts.net/jira/browse/CMC-737

File: Jenkinsfile_CNP [hunk 3]
```diff
@@ -52,6 +52,14 @@ def uploadCoreCaseDataDefinitions(env) {
   }
 }
 
+def uploadBpmnDiagrams(String env) {
+  dir("${WORKSPACE}/bin") {
```
Comment (dharmendrak) [inline Jenkinsfile_CNP:56]: can we read a directory and upload all BPMN file present there.

**Review by dharmendrak — DISMISSED** [review]

**Review by pd-dudkiewicz — DISMISSED** [review]

File: Jenkinsfile_CNP (hunk 3 above)
Comment (swalker125) [reply Jenkinsfile_CNP:56]: done

**Pull request closed without merging**